src/exporters/riemann.rs

```diff
@@ -158,64 +157,28 @@ impl Exporter for RiemannExporter {
-                attributes.insert("pid".to_string(), pid.to_string());
-
-                attributes.insert("exe".to_string(), exe.clone());
-
-                if let Some(cmdline_str) = cmdline {
-                    attributes.insert("cmdline".to_string(), cmdline_str.replace("\"", "\\\""));
-
-                    if parameters.is_present("qemu") {
-                        if let Some(vmname) = utils::filter_qemu_cmdline(&cmdline_str) {
-                            attributes.insert("vmname".to_string(), vmname);
-                        }
-                    }
-                }
-
-                // Here we define a metric name with pid + exe string suffix as riemann needs
-                // to differentiate services/metrics
-                let metric_name = format!(
-                    "{}_{}_{}",
-                    "scaph_process_power_consumption_microwatts",
-                    pid.to_string(),
-                    exe
-                );
-                if let Some(power) = topology.get_process_power_consumption_microwatts(pid) {
-                    data.push(Metric {
-                        name: metric_name,
-                        metric_type: String::from("gauge"),
-                        ttl: 60.0,
-                        hostname: get_hostname(),
-                        state: String::from("ok"),
-                        tags: vec!["scaphandre".to_string()],
-                        attributes,
-                        description: String::from("Power consumption due to the process, measured on at the topology level, in microwatts"),
-                        metric_value: MetricValueType::Text(power.to_string()),
-                    });
-                }
-            }
             // Send all data
             info!("{}: Send data", Utc::now().format("%Y-%m-%dT%H:%M:%S"));
-            for metric in metric_generator.get_metrics() {
-                rclient.send_metric(metric);
-            }
-            for metric in data {
-                rclient.send_metric(&metric);
-            }
+            metric_generator
+                .get_metrics()
+                .to_vec() // This is doing a clone of the Metric
+                .into_iter()
+                .map(|mut metric| {
+                    // Here we define a metric name with pid + exe string suffix as riemann needs
+                    // to differentiate services/metrics
+                    if metric.name == "scaph_process_power_consumption_microwatts" {
+                        metric.name = format!(
+                            "{}_{}_{}",
+                            "scaph_process_power_consumption_microwatts",
+                            metric.attributes.get("pid").unwrap(),
+                            metric.attributes.get("exe").unwrap()
+                        )
+                    }
+                    metric
+                })
+                .for_each(|metric| rclient.send_metric(&metric));
```

Comment:
Here I used the iterator because in the previous code I collected into a Vec<Metric>.
I was lazy to change and use a for loop, let me know if you prefer it because it is maybe more readable.